pkg/services/sqlstore/migrator/dialect.go

```diff
@@ -73,6 +73,9 @@ type Dialect interface {
 	Unlock(LockCfg) error
 
 	GetDBName(string) (string, error)
+
+	InsertQuery(tableName string, row map[string]any) (string, []any, error)
+	UpdateQuery(tableName string, row map[string]any, where map[string]any) (string, []any, error)
```

Comment:
rather than adding these as member functions, could/should they be a static utility function that takes a dialect?

It required reading the code for me to know if they actually do the insert or just construct a query string

Reply:
They're member functions so that the dialects can override them if needed, these just construct the query and params but we could make a version that actually performs the insert/update for situations where the dialect needed to take full control.

Reply:
Good call, I added docs and some basic tests, as well as an update so that the sql is always generated with sorted columns for consistency (in tests, but also for logging).